src/source/video_source.js

```diff
@@ -107,7 +107,7 @@ class VideoSource extends ImageSource {
     // setCoordinates inherited from ImageSource
 
     prepare() {
-        if (!this.tile || this.video.readyState < 2) return; // not enough data for current position
+        if (Object.keys(this.tiles).length === 0 || this.video.readyState < 2) return; // not enough data for current position
```

Comment:
The `CanvasSource` class also inherits from `ImageSource` and will need this change in its `prepare` method as well.